source/kernel/modules.go

```diff
@@ -42,3 +45,29 @@ func getLoadedModules() ([]string, error) {
 	}
 	return loadedMods, nil
 }
+
+func getBuiltinModules() ([]string, error) {
+	kVersion, err := getVersion()
+	if err != nil {
+		return []string{}, err
+	}
+
+	kBuiltinModPath := hostpath.LibDir.Path("modules/" + kVersion + "/modules.builtin")
+	out, err := os.ReadFile(kBuiltinModPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read file %s: %s", kBuiltinModPath, err.Error())
+	}
+
+	lines := strings.Split(string(out), "\n")
+	builtinMods := make([]string, len(lines))
+	for _, line := range lines {
+		// skip empty lines
+		if len(line) == 0 {
+			continue
+		}
```

Comment:
One suggestion: instead of checking for empty lines it would prolly be more robust to first trim all white space and then verify that the string ends with .ko (all the modules do AFAIU)

```suggestion
		line = strings.TrimSpace(line)
		if !strings.HasSuffix(line, ".ko") {
			continue
		}
```